src/update-done/update-done.c

```diff
@@ -35,11 +46,15 @@ static int save_timestamp(const char *dir, struct timespec *ts) {
                      timespec_load_nsec(ts)) < 0)
                 return log_oom();
 
-        r = write_string_file_full(AT_FDCWD, path, message, WRITE_STRING_FILE_CREATE|WRITE_STRING_FILE_ATOMIC|WRITE_STRING_FILE_LABEL, ts, NULL);
-        if (r == -EROFS)
-                log_debug_errno(r, "Cannot create \"%s\", file system is read-only.", path);
+        r = write_string_file_full(fd, ".updated", message,
+                                   WRITE_STRING_FILE_CREATE|WRITE_STRING_FILE_ATOMIC|WRITE_STRING_FILE_LABEL,
+                                   ts, NULL);
+        if (r == -EROFS && !arg_root)
+                log_debug_errno(r, "Cannot create \"%s/.updated\", file system is read-only.", dirpath);
         else if (r < 0)
-                return log_error_errno(r, "Failed to write \"%s\": %m", path);
+                return log_error_errno(r, "Failed to write \"%s/.updated\": %m", dirpath);
+        log_debug("%s/.updated updated to TIMESTAMP_NSEC="NSEC_FMT, dirpath, timespec_load_nsec(ts));
```

Comment:
this line should be behind an `else` i guess. otherwise you'll claim we made an update in the EROFS case, even though we didn't

Reply:
Good catch, thanks.